fedora/infra.py

```diff
@@ -150,31 +197,119 @@ def format_title(outage):
 
         message = f"I checked [Fedora Status]({self.fedorastatus_url}) and there are "
         if not ongoing and not planned:
-            message = message + f"**no planned or ongoing outages on Fedora Infrastructure.**{NL}"
-        else:
-            message = message + (
-                f"**{len(ongoing)} ongoing** and **{len(planned)} planned** "
-                f"outages on Fedora Infrastructure.{NL}"
-            )
-            if ongoing:
-                message = message + f"##### Ongoing{NL}"
-                for outage in ongoing:
-                    message = message + f" * {format_title(outage)}{NL}"
-                    # TODO: Make these times relative (e.g. started 1 hour ago)
-                    message = message + f"   Started at: {outage['startdate']}{NL}"
-                    message = message + (
-                        f"   Estimated to end: "
-                        f"{outage['enddate'] if outage['enddate'] else 'Unknown'}{NL}"
-                    )
+            message += f"**no planned or ongoing outages on Fedora Infrastructure.**{NL}"
+            await evt.respond(message)
+            return
+
+        message += (
+            f"**{len(ongoing)} ongoing** and **{len(planned)} planned** "
+            f"outages on Fedora Infrastructure.{NL}"
+        )
+
+        if ongoing:
+            message += f"##### Ongoing{NL}"
+            for outage in ongoing:
+                start_time = self.date_validation(outage.get("startdate"))
+                end_time = (
+                    self.date_validation(outage.get("enddate"))
+                    if outage.get("enddate") is not None
+                    else None
+                )
+
+                now = datetime.datetime.now(start_time.tzinfo)
+                delta_time = now - start_time
+
+                relative_start_time = self.get_relative_time(delta_time)
+                relative_end_time = (
+                    self.get_relative_time(end_time - now) if end_time is not None else None
+                )
+
+                message += f" * {format_title(outage)}{NL}"
+                message += f"   Started : {relative_start_time}{NL}{start_time}"
```

Comment:
I assume the `{NL}` means "newline"?

Perhaps formatting it so it reads `Started: 2024-04-01-11:30:38 (3 minutes ago)` would be clearer.